nuopc vs. gnu compiler

MOM6 is failing to compile with gnu in the nuopc driver:
```
f951: Warning: Nonexistent include directory ‘/glade/scratch/kate/Bering_JRA_gnu/bld/gnu/mpt/nodebug/nothreads/nuopc/finclude’ [-Wmissing-include-dirs]
/glade/u/home/kate/CESM/components/mom/MOM6/config_src/nuopc_driver/mom_cap.F90:356:19:

  356 |      read(value, '(i)', iostat=iostat) scalar_field_count
      |                   1
Error: Nonnegative width required in format string at (1)
/glade/u/home/kate/CESM/components/mom/MOM6/config_src/nuopc_driver/mom_cap.F90:379:19:

  379 |      read(value, '(i)', iostat=iostat) scalar_field_idx_grid_nx
      |                   1
Error: Nonnegative width required in format string at (1)
/glade/u/home/kate/CESM/components/mom/MOM6/config_src/nuopc_driver/mom_cap.F90:402:19:

  402 |      read(value, '(i)', iostat=iostat) scalar_field_idx_grid_ny
      |                   1
Error: Nonnegative width required in format string at (1)
/glade/u/home/kate/CESM/components/mom/MOM6/config_src/nuopc_driver/mom_cap.F90:1438:51:

 1438 |          scalar_field_name, scalar_field_count, rc)
      |                                                   1
Error: Type mismatch in argument ‘value’ at (1); passed REAL(16) to REAL(8)
/glade/u/home/kate/CESM/components/mom/MOM6/config_src/nuopc_driver/mom_cap.F90:1445:52:

 1445 |           scalar_field_name, scalar_field_count, rc)
      |                                                    1
Error: Type mismatch in argument ‘value’ at (1); passed REAL(16) to REAL(8)
gmake: *** [mom_cap.o] Error 1
```

--- a/config_src/nuopc_driver/mom_cap.F90
+++ b/config_src/nuopc_driver/mom_cap.F90
@@ -353,7 +353,7 @@ subroutine InitializeP0(gcomp, importState, exportState, clock, rc)
        file=__FILE__)) &
        return
   if (isPresent .and. isSet) then
-     read(value, '(i)', iostat=iostat) scalar_field_count
+     read(value, *, iostat=iostat) scalar_field_count
      if (iostat /= 0) then
        call ESMF_LogSetError(ESMF_RC_ARG_BAD, &
             msg=subname//": ScalarFieldCount not an integer: "//trim(value), &
@@ -376,7 +376,7 @@ subroutine InitializeP0(gcomp, importState, exportState, clock, rc)
        file=__FILE__)) &
        return
   if (isPresent .and. isSet) then
-     read(value, '(i)', iostat=iostat) scalar_field_idx_grid_nx
+     read(value, *, iostat=iostat) scalar_field_idx_grid_nx
      if (iostat /= 0) then
         call ESMF_LogSetError(ESMF_RC_ARG_BAD, &
              msg=subname//": ScalarFieldIdxGridNX not an integer: "//trim(value), &
@@ -399,7 +399,7 @@ subroutine InitializeP0(gcomp, importState, exportState, clock, rc)
        file=__FILE__)) &
        return
   if (isPresent .and. isSet) then
-     read(value, '(i)', iostat=iostat) scalar_field_idx_grid_ny
+     read(value, *, iostat=iostat) scalar_field_idx_grid_ny
      if (iostat /= 0) then
         call ESMF_LogSetError(ESMF_RC_ARG_BAD, &
              msg=subname//": ScalarFieldIdxGridNY not an integer: "//trim(value), &
@@ -1434,14 +1434,14 @@ subroutine InitializeRealize(gcomp, importState, exportState, clock, rc)
   !---------------------------------
 
   if (len_trim(scalar_field_name) > 0) then
-     call State_SetScalar(dble(nxg),scalar_field_idx_grid_nx, exportState, localPet, &
+     call State_SetScalar(real(nxg,ESMF_KIND_R8),scalar_field_idx_grid_nx, exportState, localPet, &
          scalar_field_name, scalar_field_count, rc)
      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
          line=__LINE__, &
          file=__FILE__)) &
          return
 
-     call State_SetScalar(dble(nyg),scalar_field_idx_grid_ny, exportState, localPet, &
+     call State_SetScalar(real(nyg,ESMF_KIND_R8),scalar_field_idx_grid_ny, exportState, localPet, &
           scalar_field_name, scalar_field_count, rc)
      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
          line=__LINE__, &

